src/components/InternalTransfer/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { DropdownComponent, Decimal } from '../';
+import { isUsernameEnabled } from '../../api';
+import { CloseIcon } from '../../assets/images/CloseIcon';
+import { Modal } from '../../components';
+import { createInternalTransfersFetch, selectUserInfo, selectWallets, walletsFetch } from '../../modules';
+import { InternalTransferInput } from './InternalInput';
+
+export const InternalTransferComponent = () => {
+    const { formatMessage } = useIntl();
+
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const wallets = useSelector(selectWallets);
+    const user = useSelector(selectUserInfo);
+
+    const [username, setUsername] = useState('');
+    const [currency, setCurrency] = useState('');
+    const [amount, setAmount] = useState('');
+    const [otp, setOtp] = useState('');
+
+    const [show, setShow] = useState(false);
+
+    const BALANCE_FETCH_TIMEOUT = 2000;
+
+    React.useEffect(() => {
+        if (!wallets.length) {
+            dispatch(walletsFetch());
+        }
+    }, [dispatch, wallets]);
+
+    const walletsList = wallets.length ? wallets.map(item => item.currency && item.currency.toUpperCase()) : [];
+
+    const wallet = wallets.length && wallets.find(item => item.currency.toLowerCase() === currency.toLowerCase());
+
+    const balanceError = useMemo(() => classnames('cr-internal-transfer__group--balance', {
+        'cr-internal-transfer__group--error': wallet && wallet.balance && +wallet.balance < +amount,
+    }), [amount, wallet]);
+
+    const translationUsername = isUsernameEnabled() ? 'username' : 'uid';
+
+    const handleCreateTransfer = useCallback(() => {
+        const payload = {
+            currency: currency.toLowerCase(),
+            username_or_uid: username,
+            amount,
+            otp
+        };
+
+        dispatch(createInternalTransfersFetch(payload));
+        setShow(false);
+        setUsername('');
+        setCurrency('');
+        setAmount('');
+        setOtp('');
+        setTimeout(() => {  dispatch(walletsFetch()) }, BALANCE_FETCH_TIMEOUT);
+    }, [username, otp, amount, currency, dispatch]);
+
+    const translate = useCallback((id: string) => formatMessage({ id: id }), [formatMessage]);
+
+    const handleNavigateTo2fa = useCallback((enable2fa: boolean) => {
+        history.push('/security/2fa', { enable2fa });
+    }, []);
+
+    const handleChangeShow = () => {
+        setShow(false);
+        setUsername('');
+        setCurrency('');
+        setAmount('');
+        setOtp('');
+    }
+
+    const renderFooter = useMemo(() => {
+        return (
+            <Button
+                block={true}
+                type="button"
+                onClick={handleCreateTransfer}
+                size="lg"
+                variant="primary"
+                disabled={!username || !otp || !amount || !currency}
```

Comment:
Define function for checks and re-use it on line 165:
`!username || !otp || !amount || !currency`

Reply:
fixed